### DISCUSSION TITLE
AsyncWebServer_RP2040W missing some code

### DISCUSSION THREAD

**Discussion by palmerr23** [earlephilhower/arduino-pico#1789] [Q&A]
Hi,

I'm porting some working code from ESP32 to Pico W.

It appears that Khoi Hoang has left a few lines out of his port of ESPAsyncWebserver, specifically those implementing 

```
void send_P(int code, const String& contentType, const uint8_t * content, size_t len,
                AwsTemplateProcessor callback = nullptr);
void send_P(int code, const String& contentType, PGM_P content, AwsTemplateProcessor callback = nullptr);
```

all that is required to fix the issue is adding 
```
void AsyncWebServerRequest::send_P(int code, const String& contentType, const uint8_t * content, size_t len, AwsTemplateProcessor callback){
  send(beginResponse_P(code, contentType, content, len, callback));
}

void AsyncWebServerRequest::send_P(int code, const String& contentType, PGM_P content, AwsTemplateProcessor callback){
  send(beginResponse_P(code, contentType, content, callback));
}
```
somewhere around line 1200 in AsyncWebRequestt_RP2040W.cpp

Khoi Hoang has archived the repository, so I can't raise a PR to fix it. Any ideas about whether someone else has taken over maintenance for this library?

**Comment by earlephilhower** [comment]
Have you tried asking @khoih-prog why he archived it?  Maybe you can just fork and fix and be the new keeper.

**Comment by palmerr23** [comment]
Earle,

He has Discussions turned off on all his repos (even the unarchived ones), and I can't find any other way to contact him.

There don't seem to be any other public forks, so I have forked and fixed, and will wait and see if it attracts any traffic.
